tasmota/xsns_73_hp303b.ino

```diff
@@ -0,0 +1,161 @@
+/*
+  xsns_72_hp303b.ino - HP303B digital barometric air pressure sensor support for Tasmota
+
+  Copyright (C) 2020  Theo Arends
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#ifdef USE_I2C
+#ifdef USE_HP303B
+/*********************************************************************************************\
+ * HP303B - Gas (TVOC - Total Volatile Organic Compounds) and Air Quality (CO2)
+ *
+ * Source: Lolin LOLIN_HP303B_Library
+ *
+ * I2C Address: 0x77 or 0x76
+\*********************************************************************************************/
+
+#define XSNS_73 73
+#define XI2C_52 52 // See I2CDEVICES.md
+
+#define HP303B_ADDR1 0x77
+#define HP303B_ADDR2 0x76
+
+#include <LOLIN_HP303B.h>
+// HP303B Opject
+LOLIN_HP303B HP303BSensor = LOLIN_HP303B();
+
+struct HP303BDATA
+{
+  uint8_t address;
+  uint8_t addresses[2] = {HP303B_ADDR1, HP303B_ADDR2};
+  uint8_t type = 0;
+  uint8_t valid = 0;
+  int32_t temperature;
+  int32_t pressure;
+  int16_t oversampling = 7;
+  char types[7] = "HP303B";
+} HP303B;
+
+/*********************************************************************************************/
+
+bool HP303B_Read(int32_t &t, int32_t &p, uint8_t hp303b_address)
+{
+  HP303BSensor.begin(hp303b_address);
+
```

Comment:
the `HP303BSensor.begin` should only do by init phase of the sensor, not every read.

Reply:
Moved it to detect